### DISCUSSION TITLE
[Feature] Select Multiple faces for hiding or tagging in the photo information panel

### DISCUSSION THREAD

**Discussion by yewkay** [immich-app/immich#4825] [Feature Request]
### The feature

In a photo of a crowd with many faces, it would be useful to allow multiple selection on the infomation panel.
[Sample photo](https://imgur.com/a/FsArrcs)

We can either select multiple faces for hiding or to tag them in the small thumbnails itself.

1 way to implement can be to add the 3 dots on the upper right and a circle with tick for selection(similar to the main gallery page) 

Clicking on the 3 dots will show a menu to allow for some features like:
1. tagging,
2. untagging or changing wrongly auto tagged names
3.  hiding individual faces or,
4. any feature in future. 
 
Clicking on the circle with a tick will select the photo so that multiple faces can be selected for hiding.

Thank you!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Hypnos-69** [comment]
Hello,

I have the same requirement, and I’m surprised that this feature hasn’t attracted more interest since 2023...

Here’s my take on the matter 🙂.

## Feature Description

When opening a photo with a large number of detected faces (for example: crowd photos, museums, conventions, or special effects exhibitions with multiple masks/busts), managing false positives or unwanted faces is currently very tedious.

As of now, users have to click the "Hide" or "Delete" button individually for every single face in the right-hand sidebar ("Modifier les visages" / "Edit faces"). If a photo contains dozens of false positives, this requires dozens of repetitive clicks, to click on the bin icon next to a face, and then to confirm the action.

## Solution / Implementation Idea

I would love to see an option to perform bulk actions directly inside the "Edit faces" sidebar of a single asset.

This could include:
 - **_Checkboxes_** on each detected face thumbnail in the sidebar.
 - A "**_Select All" / "Deselect All_**" toggle at the top of the list.
 - **_Bulk actions buttons_** at the top or bottom of the sidebar once multiple faces are selected, allowing the user to either:
	- **_Hide_** all selected faces.
	- **_Delete_** all selected faces from the asset.
	- Apply a **_Filter_** that would automatically deselect faces that have already associated with people.

## Additional Context

Here is an example of a screenshot where a single picture contains dozens of detected faces (in this case, special effects creature busts) that need to be cleared out.

<img width="1875" height="917" alt="2026-06-09_09h26_39" src="https://github.com/user-attachments/assets/0e84efcd-dc1f-4d4f-a570-91c24d1ee3e7" />


A second AI-generated image that gives an idea of what the face editing panel will look like with these new features.

<img width="1475" height="720" alt="Gemini_Generated_Image_poqrp2poqrp2poqr" src="https://github.com/user-attachments/assets/cf3d6ae0-03e3-4da7-927c-c552f7b7afb4" />

A huge thanks to the Immich team !!!

**Comment by LinkeLoutje** [comment]
Another surprise here! Quite a similar problem. Maybe a different solution.

When you select a face from the people page. I notice immediately that all bundled images are from an unknown person. I'd like to be able to delete all these faces from the image from that top page, instead of: clicking the individual image, opening the info pane, click edit in the people module, delete a face.
